### ISSUE TITLE
Evaluate addition of a Router

### ISSUE THREAD

**Issue by shrugs** [urbit/bridge#148]
One of the goals of bridge is to be easily offline-capable with a simple HTTP server, so traditional browser history using paths won't work: simple routers like python's built-in won't know to render `index.html` when it receives `/some-route`. The solution to this is a memory or hash router, which [React Router](https://reacttraining.com/react-router/web/guides/quick-start) provides and [Reach Router](https://reach.tech/router) can support with a [few lines of glue code](https://github.com/will-stone/hash-source/blob/master/src/createHashSource.ts).

Changing to an established router with proper history sync will probably solve bugs like

- https://github.com/urbit/bridge/pull/86#issuecomment-475897713
- https://github.com/urbit/bridge/issues/136
- https://github.com/urbit/bridge/issues/147

And we will soon need a way to react to initial URL state, for example from an invite link, which we get for free with a declarative router.

One problem, though, is that both projects are undergoing a bunch of change: [Reach Router is stopping at 1.0 (with bugfixes) and being merged into React Router in v5.x](https://reacttraining.com/blog/reach-react-router-future/). The milestone for this is [here](https://github.com/ReactTraining/react-router/issues/6497).

This means it's probably a bad idea to refactor our app right away, but something we should definitely consider once React Router 5.x ships with the smaller bundle size and friendlier API.

cc @Fang- thoughts?

**Comment by Fang-** [comment]
Huh, I drafted a response to this at some point but apparently never sent it. iirc the gist of it was "ugh dependencies" but also "we need less bugs". I don't have strong opinions here besides going with the simplest thing that Just Works. We can talk more seriously about crossing this bridge (heh) once React Router 5 lands.

**Comment by shrugs** [comment]
@Fang- agreed. went with the "make our home-grown router better" approach for indigo, but will consider moving to the new react router once it ships

**Comment by shrugs** [comment]
this will also let us correctly use unload blockers to get the 'are you sure you want to leave' dialog UX during transactions

**Comment by vvisigoth** [comment]
What's the status of this?

**Comment by Fang-** [comment]
Looks like react-router v5 has shipped, meaning we could go and swap it in. I think it's just a matter of updating `/lib/useRouter`?

**Comment by shrugs** [comment]
the hooks update is in a 5.x minor, so up to you

**Comment by vvisigoth** [comment]
I don't care. Leave it up to @Fang-
—
~poldec-tonteg
http://urbit.org

**Comment by Fang-** [comment]
Pretty sure work on this has been done. I don't think we've seen very many routing bugs. Closing.
